dashboard/modules/job/utils.py

```diff
@@ -64,3 +98,119 @@ def file_tail_iterator(path: str) -> Iterator[Optional[List[str]]]:
                 lines = []
                 chunk_char_count = 0
                 curr_line = None
+
+
+async def parse_and_validate_request(req: Request, request_type: dataclass) -> Any:
+    """Parse request and cast to request type. If parsing failed, return a
+    Response object with status 400 and stacktrace instead.
+    """
+    import aiohttp
+
+    try:
+        return validate_request_type(await req.json(), request_type)
+    except Exception as e:
+        logger.info(f"Got invalid request type: {e}")
+        return Response(
+            text=traceback.format_exc(),
+            status=aiohttp.web.HTTPBadRequest.status_code,
+        )
+
+
+async def get_driver_jobs(
+    gcs_job_info_stub: gcs_service_pb2_grpc.JobInfoGcsServiceStub,
+) -> Tuple[Dict[str, JobDetails], Dict[str, DriverInfo]]:
+    """Returns a tuple of dictionaries related to drivers.
+
+    The first dictionary contains all driver jobs and is keyed by the job's id.
+    The second dictionary contains drivers that belong to submission jobs.
+    It's keyed by the submission job's submission id.
+    Only the last driver of a submission job is returned.
+    """
+    request = gcs_service_pb2.GetAllJobInfoRequest()
+    reply = await gcs_job_info_stub.GetAllJobInfo(request, timeout=5)
```

Comment:
Does this only contain the running jobs or it contains everything? Will this become too big as time going?

Reply:
This part of the code was introduced by this [PR](https://github.com/ray-project/ray/pull/25902), and I don't know the context. My guess, is cause we only have `GcsJobManager::HandleGetAllJobInfo` but don't implement `GcsJobManager::HandleGetJobInfo`